OpenEdX/Managers/DeepLinkManager/DeepLinkManager.swift

```diff
@@ -0,0 +1,71 @@
+//
+//  DeepLinkManager.swift
+//  OpenEdX
+//
+//  Created by Anton Yarmolenka on 24/01/2024.
+//
+
+import Foundation
+import Core
+import UIKit
+
+public protocol DeepLinkService {
+    func configureWith(launchOptions: [UIApplication.LaunchOptionsKey: Any]?)
+    func handledURLWith(app: UIApplication, open url: URL, options: [UIApplication.OpenURLOptionsKey: Any]) -> Bool
+}
+
+class DeepLinkManager {
+    private var service: DeepLinkService?
+    
+    // Init manager
+    public init(config: ConfigProtocol) {
+        self.service = self.serviceFor(config: config)
```

Comment:
The DeeplinkManager can register just one service Branch service. How about configuring it to register multiple services thoughts?